jbmc/src/java_bytecode/assignments_from_json.cpp

```diff
@@ -0,0 +1,837 @@
+/*******************************************************************\
+
+Module: Assignments to values specified in JSON files
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include "assignments_from_json.h"
+
+#include "ci_lazy_methods_needed.h"
+#include "java_static_initializers.h"
+#include "java_string_library_preprocess.h"
+#include "java_string_literals.h"
+#include "java_utils.h"
+
+#include <goto-programs/class_identifier.h>
+#include <util/allocate_objects.h>
+#include <util/array_element_from_pointer.h>
+#include <util/expr_initializer.h>
+#include <util/prefix.h>
+#include <util/unicode.h>
+
+/// Values passed around between most functions of the recursive deterministic
+/// assignment algorithm entered from \ref assign_from_json.
+/// The values in a given `det_creation_infot` are never reassigned, but the
+/// ones that are not marked `const` may be mutated.
+struct det_creation_infot
```

Comment:
Suggest naming this `object_creation_infot`, `det`, presumably short for "deterministic", isn't really used in cbmc apart from its antonym "nondet".

Reply:
Changed it to `object_creation_infot`. Similarly, I also renamed `det_creation_referencet` (in `assignments_from_json.h`) to `object_creation_referencet`.